core/BCD.savi

```diff
@@ -0,0 +1,116 @@
+:: Binary Coded Decimal (BCD) representation of an U64 integer.
+:: 
+:: Each digit is represented by a nibble (4 bits), requiring
+:: a total of 80 bits for up to 20 digits of the U64.
+:: Digits are stored in reverse order, so the the decimal `123`
+:: is internally stored as `0x321`.
+::
+:: In addition, we store a termination nibble `0xF` after the last
+:: digit, so decimal `123` is internally stored `0xF321`.
+:: At bit position 58 of the high word, we store the number of digits
+:: (requires 5 bits).
+::
+:: TODO: The code would become trivial if we had a `U128` type or a
+:: bit-shift connected pair of two U64.
+::
+:struct val BCD.U64
```

Comment:
I'm hesitant to take up `BCD` in the global/core namespace in case there is some other acronym that user programs want to use this name for.

Can I suggest calling this `U64.BCD` instead of `BCD.U64`? I think it's still clear, and avoids taking up a new top-level name.

Reply:
Yes, that's perfect. Though, it's quite easily possible to make it also store sign information to also cover `I64`. Though I am not sure how `BCD` can be useful other than for generating a decimal string representation.

Reply:
If we think it's not useful to users, we could just make it a private type instead. We could always make it public later if it turns out to be useful for some other use case.